### PR TITLE
-j flag for mjolnir exes

### PR DESCRIPTION
**Pull request by nilsnolde**
a much simpler version of #4151

### CODE REVIEW

File: valhalla/baldr/graph_utils.h [hunk 1]
```diff
@@ -6,14 +6,14 @@
 
 #include <boost/property_tree/ptree.hpp>
 
-#include "baldr/graphid.h"
-#include "baldr/graphreader.h"
-#include "baldr/graphtile.h"
-#include "filesystem.h"
-#include "midgard/logging.h"
+#include <valhalla/baldr/graphid.h>
```
Comment (nilsnolde) [inline valhalla/baldr/graph_utils.h:9]: moved this header from mjolnir to baldr and called it graph_utils.h. it was only used in valhalla_add_elevation.cc (& its test), but could be useful in other places and really shouldn't live inside mjolnir

File: src/mjolnir/valhalla_add_elevation.cc [hunk 2]
```diff
@@ -30,90 +34,97 @@ enum Input { CONFIG, TILES };
...
-  options.add_options()("h,help",
-                        "Print usage")("c,config", "Path to the configuration file.",
-                                       opt::value<
-                                           std::string>())("t,tiles", "Tiles to add elevations to",
-                                                           opt::value<std::vector<std::string>>());
-
-  options.allow_unrecognised_options();
-  std::string config_file;
-  std::vector<std::string> tiles;
+std::deque<GraphId> get_tile_ids(const boost::property_tree::ptree& pt,
```
Comment (nilsnolde) [inline src/mjolnir/valhalla_add_elevation.cc:37]: this is just one bigger refactor I felt like doing, to make this exe somewhat consistent with all others.

in a follow up PR I'll actually break out all of this boilerplate arg parsing to a private header

File: src/mjolnir/valhalla_convert_transit.cc [hunk 3]
```diff
@@ -63,11 +69,6 @@ bool ParseArguments(int argc, char* argv[]) {
       std::sort(onestoptests.begin(), onestoptests.end());
     }
 
-    if (result.count("target_directory")) {
-      pt.get_child("mjolnir").erase("tile_dir");
-      pt.add("mjolnir.tile_dir", result["target_directory"].as<std::string>());
-    }
```
Comment (nilsnolde) [inline src/mjolnir/valhalla_convert_transit.cc:69]: this was duplicated code, already exists a few lines up

**Review by kevinkreiser — DISMISSED** [review]
looks good just would like to keep the elevation tile_ids things as an unordered_set

**Comment by nilsnolde** [comment]
> looks good just would like to keep the elevation tile_ids things as an unordered_set

did that and merged it into the superseding PR. closing here

**Pull request closed without merging**